nexus/src/db/datastore/disk.rs

```diff
@@ -64,19 +67,29 @@ impl DataStore {
 
         let gen = disk.runtime().gen;
         let name = disk.name().clone();
-        let disk: Disk = diesel::insert_into(dsl::disk)
-            .values(disk)
-            .on_conflict(dsl::id)
-            .do_nothing()
-            .returning(Disk::as_returning())
-            .get_result_async(self.pool())
-            .await
-            .map_err(|e| {
+        let project_id = disk.project_id;
+
+        let disk: Disk = Project::insert_resource(
+            project_id,
+            diesel::insert_into(dsl::disk)
+                .values(disk)
+                .on_conflict(dsl::id)
+                .do_nothing(),
+        )
+        .insert_and_get_result_async(self.pool())
+        .await
+        .map_err(|e| match e {
+            AsyncInsertError::CollectionNotFound => Error::ObjectNotFound {
```

Comment:
Mentioned elsewhere, but it's surprising there's no authz check here.  If there were, you could use `authz_disk.not_found()` to generate the `ObjectNotFound` error.